### PR TITLE
support String#[pos, len]= val

### PR DESCRIPTION
**Pull request by takahashim**
cf. http://docs.ruby-lang.org/en/2.2.0/String.html#method-i-5B-5D-3D

### CODE REVIEW

File: mrblib/string.rb [hunk 1]
```diff
@@ -154,13 +154,48 @@ def each_byte(&block)
...
+          a = self[(posnum + pos.length)..-1]
+          self.replace([b, value, a].join(''))
+          return value
+        else
+          raise IndexError, "string not matched"
+        end
+      else
+        pos += self.length if pos < 0
+        if pos < 0 || pos > self.length
+          raise IndexError, "index #{args[0]} out of string"
```
Comment (franckverrot) [inline mrblib/string.rb]: do you think we could we rephrase “... out of bounds” ?

Comment (takahashim) [inline mrblib/string.rb]: I'm not sure which is better, but it' from CRuby:

```ruby
$ ruby -e '"foo"[4,1]="X"'
-e:1:in `[]=': index 4 out of string (IndexError)
	from -e:1:in `<main>'
```

Comment (franckverrot) [inline mrblib/string.rb]: Yes, I figured :-)

On a somehow related note, I can't see where in the spec `String#[]` (15.2.10.5.6) is supposed to raise anything in those cases. :-/

Comment (takahashim) [inline mrblib/string.rb]: I guess `String#[]` and `String#[]=` are different methods, and the latter is not defined in ISO Ruby :worried:

Comment (franckverrot) [inline mrblib/string.rb]: Whoops... indeed :+1:

**Pull request merged**